Feature/updated installation script

JaytidaA opened the pull request:
Solves issue #164

+ Added a check for the specific Linux distro using /etc/os-release
+ Added instructions for Arch Linux
+ Sets the hardware clock to /Etc/UTC
+ Installs sudo and sets it up on Arch Linux
+ Sets up the dialout group if it does not exist on Arch Linux
+ Installs the required libraries
+ Resets the hardware clock to initial value

NachtSpyder04 reviewed (changes requested):
Format the code properly

JaytidaA commented:
> Format the code properly

Done.

JaytidaA commented:
Is there anything else that needs to be done?

NachtSpyder04 reviewed (approved):
LGTM!

JaytidaA commented:
Latest changes:
+ Added Arch Linux Latest container to the `test-install-script.yml` workflow (this helped tremendously)
+ Updated the version of the "checkout" repository to version 4 in the two .yml files which use it.

# Installation script updates after adding the archlinux container to GitHub actions:
+ Changed all instances of `pacman` to make use of the `noconfirm` flag so that installation is not interrupted for every use of pacman.
+ Set up a separate if clause for when CI is used for Arch Linux as it requires that the machine be rebooted to update permissions but we cannot do that on Github Actions. It does not care for installation of sudo, changing timezones or even adding the current user to the group.
+ Since `sudo` is not installed on Arch Linux, by default. If a user who does not have it installed runs the script, `sudo` is first installed on their system and then they are prompted to rerun the script after rebooting.
+ The script for the users of Arch Linux not in a CI environment works fine as well and was tested by me in a fresh install of Arch by me.
+ Added `echo` logs for which Operating System is being used currently.

Shankari02 commented:
![image](https://github.com/user-attachments/assets/ac678d1f-5533-4346-bf7a-4003157cf8ff)
@JaytidaA please fix these two warnings

Shankari02 reviewed (approved):
:shipit:
